### PR TITLE
Identify spark pod with label spark-app-selector

### PR DESCRIPTION
**Pull request by yangwwei**
Native spark on K8s adds a label "spark-app-selector" to track spark app ID. We should directly read from this label.

### CODE REVIEW

**Comment by yangwwei** [comment]
recheck

File: pkg/dispatcher/dispatcher_test.go [hunk 1]
```diff
@@ -17,13 +17,15 @@ limitations under the License.
...
 import (
+	"fmt"
 	"runtime"
 	"strings"
 	"sync"
 	"sync/atomic"
 	"testing"
 	"time"
 
+	"github.com/apache/incubator-yunikorn-k8shim/pkg/common/utils"
```
Comment (wilfred-s) [inline pkg/dispatcher/dispatcher_test.go]: all yunikorn imports should be grouped together. I filed new issues to fix the linter setting but this should be with the events import (k8shim  #75)

File: pkg/dispatcher/dispatcher_test.go [hunk 2]
```diff
@@ -201,31 +204,57 @@ func TestDispatchTimeout(t *testing.T) {
 	}()
 
 	// pretend to be an time-consuming event-handler
```
Comment (wilfred-s) [inline pkg/dispatcher/dispatcher_test.go]: This comment does not make sense any more as the sleep has been removed

File: pkg/dispatcher/dispatcher_test.go [hunk 3]
```diff
@@ -201,31 +204,57 @@ func TestDispatchTimeout(t *testing.T) {
...
+	event2 := TestAppEvent{
+		appID:     "test-02",
+		eventType: events.RunApplication,
+		flag:      make(chan bool),
+	}
+
+	// dispatch all events
+	Dispatch(event0)
+	Dispatch(event1)
+	Dispatch(event2)
```
Comment (wilfred-s) [inline pkg/dispatcher/dispatcher_test.go]: Why unwrap the for loop? A loop makes the code much more readable and easier to maintain.

File: pkg/dispatcher/dispatcher_test.go (hunk 1 above)
Comment (yangwwei) [reply pkg/dispatcher/dispatcher_test.go]: Done.

File: pkg/dispatcher/dispatcher_test.go (hunk 2 above)
Comment (yangwwei) [reply pkg/dispatcher/dispatcher_test.go]: Fixed.

File: pkg/dispatcher/dispatcher_test.go (hunk 3 above)
Comment (yangwwei) [reply pkg/dispatcher/dispatcher_test.go]: This is no longer needed. Earlier I need to access each of the events to verify something. But that is now removed. I just changed to use a simple loop. Thanks.

**Comment by yangwwei** [comment]
Thanks @TaoYang526 for the review.
Since the change is trivial, I have merged the PR after addressing @wilfred-s' comment, because right now the UT is broken without this fix. @wilfred-s , if there is any more comment not been addressed, we can use a separate issue to track. Hope this makes sense.

**Pull request merged**